crates/core/src/operations/optimize.rs

```diff
@@ -251,10 +262,35 @@ impl<'a> OptimizeBuilder<'a> {
         self.min_commit_interval = Some(min_commit_interval);
         self
     }
+
+    /// Commit Writes
+    pub fn with_commit_writes(mut self, commit_writes: bool) -> Self {
+        self.commit_writes = commit_writes;
+        self
+    }
+}
+
+impl<'a> OptimizeBuilder<'a> {
+    /// Commit writes after processing
+    pub async fn commit_writes(
+        self,
+        commit_info: &CommitInfo
```

Comment:
do you still need to use the commit info after the write? if not, we can pass it in as value, not a reference to reduce memory copy.